### DISCUSSION TITLE
Allow File Storage on IPFS

### DISCUSSION THREAD

**Discussion by META-DREAMER** [directus/directus#17612] [Feature Requests]
### Summary

Would be great to have an option to store and read files from IPFS natively in Directus. Its somewhat possible to simulate this today in Directus using something like [Filebase's S3 compatible IPFS storage service](https://docs.filebase.com/), but when getting download links to assets etc it doesn't go through IPFS. 

### Motivation

IPFS support would allow for cheap efficient / decentralized storage, making Directus a lot more web3 friendly.

### Detailed Design

Details on interacting with files on IPFS can be found here: https://docs.filebase.com/getting-started/ipfs-getting-started-guide#filebase-ipfs-pinning-service-api

Guide on using js-ipfs client to add and read files from IPFS: https://proto.school/regular-files-api

Using just filebase with the S3 API and then storing the IPFS CIDs in Directus metadata could be a good low-effort solution, can reuse existing code for s3 file storage and just implement IPFS support for fetching assets.

### Must Have:
- Ability to automatically store CIDs for IPFS files in Directus Files table when using Filebase S3 API

### Should Have:
- Ability to return an IPFS gateway URL instead of Directus /assets/ URL when downloading / accessing a file

### Could Have:
- Native ability to upload directly to IPFS without using Filebase's S3 API, making it work with any IPFS backend

### Won't Have:
- Running an IPFS node on Directus instance itself

### Drawbacks

 Slight platform dependency on Filebase if we do the easy implementation route using existing S3 API. Not any worse than e.g. Cloudinary integration, but not as impactful as a native IPFS implementation.

If we want to avoid Filebase vendor lock-in, a lot more work to write new code to handle native upload / fetch with IPFS.

### Alternatives

Could start with the Filebase S3 implementation as a starting point and then later implement native IPFS uploads.

### Adoption Strategy

Not a breaking change, can be adopted simply as a new file storage provider which can be configured alongside existing providers users might have.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
